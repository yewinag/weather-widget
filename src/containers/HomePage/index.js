import React, { Component } from 'react';
import { connect } from 'react-redux';

import TypeSearch from '../../components/TypeSearch';
import WeatherBoard from '../../components/WeatherBoard';

import { fetchWeather } from '../../actions/weatherActions';
class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            city: ''
        };
    }
     
    render() { 
        return (
            <React.Fragment>
                <TypeSearch 
                    onChange={(value)=> this.setState({ city: value})}
                    onSearch={()=> this.props.fetchWeather(this.state.city)}
                />                
                <WeatherBoard />
            </React.Fragment>
        );
    }
}
 
const mapDispatchToProps = (dispatch) => {
    return {
        fetchWeather: (params) => dispatch(fetchWeather(params))                    
    }
}
const mapStateToProps = (state) => {
    return {
        weather: state.weather
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(HomePage)