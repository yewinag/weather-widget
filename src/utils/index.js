
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../reducers';
import { middlewares } from '../store';

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};