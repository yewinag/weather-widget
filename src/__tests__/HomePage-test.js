import React from 'react';
import { Provider } from 'react-redux';
import { testStore } from '../utils';
import HomePage from '../containers/HomePage';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('Home page container', () => {
    let store;
    beforeEach(() => {
        store = testStore();
    });   
    
    it('render home page container', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <HomePage />                
            </Provider>
        )
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});