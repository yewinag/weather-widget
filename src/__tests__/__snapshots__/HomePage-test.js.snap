// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page container render home page container 1`] = `ShallowWrapper {}`;
